pkcs11/_pkcs11.pyx

```diff
@@ -109,6 +109,8 @@ cdef class MechanismWithParam:
         cdef CK_RSA_PKCS_OAEP_PARAMS *oaep_params
         cdef CK_RSA_PKCS_PSS_PARAMS *pss_params
         cdef CK_ECDH1_DERIVE_PARAMS *ecdh1_params
+        cdef CK_KEY_DERIVATION_STRING_DATA *aes_ecb_params
+        cdef CK_AES_CBC_ENCRYPT_DATA_PARAMS *aes_cbc_params
```

Comment:
This is getting to be a good indication that the mechanism types above should be implemented as subclasses of MechanismWithParam (which should be renamed), `mechanism_param` keyword should be removed and this class allowed as a parameter to `mechanism`.

This would allow `mechanism=RSA_PKCS`, `mechanism=Mechanism(AES_CBC, param=iv)` or `mechanism=mechanisms.RSA_PKCS_OAEP(params=...)` with proper typing and a much better point for implementing extensions beyond this package.

That doesn't have to be done now, just a note.

Reply:
So shall this be added as a feature request, or do you want to take this up now?

If one has been created, I could add its link here in order to complete the tracking of this suggestion.

Reply:
If you wanted to give refactoring it a go, I'd be keen. I've opened #98 to track it.

Reply:
hi Danni, if it's okay with you, I'll attempt that once these changes have been merged.